Detectors/TRD/base/src/PadPlane.cxx

```diff
@@ -158,3 +162,59 @@ void PadPlane::setNrows(int n)
   }
   mNrows = n;
 };
+
+double PadPlane::getPadRow(double z) const
+{
+  double lengthCorr = mLengthIPad / mLengthOPad;
+
+  // calculate position based on inner pad length
+  double padrow = -z / mLengthIPad + mNrows / 2;
+
+  // correct row for outer pad rows
+  if (padrow <= 1.0) {
+    padrow = 1.0 - (1.0 - padrow) * lengthCorr;
+  }
+
+  if (padrow >= double(mNrows - 1)) {
+    padrow = double(mNrows - 1) + (padrow - double(mNrows - 1)) * lengthCorr;
+  }
+
+  // sanity check: is the padrow coordinate reasonable?
+  if (padrow < 0.0 || padrow > double(mNrows)) {
+    LOG(error) << "ERROR: hit with z=" << z << ", padrow " << padrow
+               << " outside of chamber";
+  }
+
+  return padrow;
+}
+
+double PadPlane::getPad(double y, double z) const
+{
+  int padrow = getPadRow(z);
+  double padrowOffset = getPadRowOffsetROC(padrow, z);
+  double tiltOffsetY = getTiltOffset(padrow, padrowOffset);
+  // double tiltOffsetY = getTiltOffset(padrowOffset);
+
+  double pad = y / mWidthIPad + mNcols / 2.;
+
+  double lengthCorr = mWidthIPad / mWidthOPad;
+  // correct row for outer pad rows
+  if (pad <= 1.0) {
+    pad = 1.0 - (1.0 - pad) * lengthCorr;
+  }
+
+  if (pad >= double(mNcols - 1)) {
+    pad = double(mNcols - 1) + (pad - double(mNcols - 1)) * lengthCorr;
+  }
+
+  double tiltOffsetPad;
+  if (pad <= 1.0 || pad >= double(mNcols - 1)) {
+    tiltOffsetPad = tiltOffsetY / mWidthOPad;
+    pad += tiltOffsetPad;
+  } else {
+    tiltOffsetPad = tiltOffsetY / mWidthIPad;
```

Comment:
Same here, define `mInverseWidthIPad` from `mWidthIPad`.

Reply:
I would not put too much effort into optimizing the code at this level. Compilers should be pretty smart these days, and the limit is probably not floating point performance, but memory access patterns. Let's try to keep the code readable, and optimize only if we actually see the need for speed.
Remember Donald Knuth's words: Premature optimization is the source of all evil ;-)

Reply:
I wouldn't call this premature optimization, but a good practice that we did follow elsewhere in several places. This is just replacing divisions with multiplications whenever it's possible to have the inverse of one of the factors precalculated. So basically, this is a free lunch. The compiler will optimize integer division but won't do it consistently with floating-point numbers in the denominator.